panic/src/lib.rs

```diff
@@ -5,3 +5,4 @@
 #![no_std]
 
 mod panicking;
+pub mod thread;
```

Comment:
This `thread.rs` is in the `mc-sgx-panic` crate.
[thread/mod.rs](https://github.com/rust-lang/rust/blob/master/library/std/src/thread/mod.rs), `panicking.rs`, and `panic.rs` are all in `::std`. `panicking.rs` is a private module in `::std`
```mermaid
flowchart LR
    A["panic.rs"] --> B["thread.rs"]
    A --> C["panicking.rs"]
    B --> C
```
So for SGX `thread.rs` feels a bit like an outlier unless we make an `mc-sgx-std` crate.
My initial thought was to add `thread.rs` to the future `mc-sgx-sync` crate, that will require making `panicking.rs` public and _could_ result in a circular dependency if we have `panic.rs` call `thread::panicking()`, but I think we could make `panic.rs` call `panicking::panicking()` instead as a minor devation from the `::std` implementation